tasks/omnibus.py

```diff
@@ -121,9 +122,9 @@ def get_omnibus_env(
     env_override = ['INTEGRATIONS_CORE_VERSION', 'OMNIBUS_RUBY_VERSION']
     for key in env_override:
         value = os.environ.get(key)
-        # Only overrides the env var if the value is a non-empty string.
+        # Only overrides the env var taking the effective SHA-1 if the value is a non-empty string.
         if value:
-            env[key] = value
+            env[key] = subprocess.check_output(("git", "rev-parse", "--verify", value), text=True).rstrip()
```

Comment:
This only resolves the reference if the value got through the environment, and not via release.json (from [line 99](https://github.com/DataDog/datadog-agent/blob/b1469e6cec26d68520c8da0b61b269b77b2bca80/tasks/omnibus.py#L99)), I think.

Reply:
Maybe something like this could work here:

```python
    for key in env_override:
        value = os.environ.get(key) or env[key]
        env[key] = subprocess.check_output(("git", "rev-parse", "--verify", value), text=True).rstrip()
```

Though in that case I would probably also rename `env_override` to something else (`external_repo_refs`?).